superset/security.py

```diff
@@ -127,12 +127,11 @@ def all_datasource_access(self):
         return self.can_access(
             'all_datasource_access', 'all_datasource_access')
 
+    def all_database_access(self):
+        return self.can_access('all_database_access', 'all_database_access')
+
     def database_access(self, database):
-        return (
-            self.can_access(
-                'all_database_access', 'all_database_access') or
-            self.can_access('database_access', database.perm)
-        )
+        return self.can_access('database_access', database.perm)
```

Comment:
Can we make this:
`return self.all_database_access() or self.can_access('database_access', database.perm)` ?

Reply:
Wondering if we should also checkout for `all_datasource_access` (which is redundant with `all_database_access`) here for backwards compatibility reasons. Say if someone setup a local role `Users with access to all data` that has `all_datasource_access`, I feel like they should see all databases when opening SQL Lab.

Reply:
BTW I feel like I wrote/merged this same PR a few months back but got reverted because of a perm issue (Alpha couldn't see the database list in SQL Lab).

PR that got reverted:
https://github.com/apache/incubator-superset/pull/7009

Related fix:
https://github.com/apache/incubator-superset/pull/7271

Reply:
Thanks for checking this out--the PR has been updated. Regarding the last comment about a similar PR having gone in and gotten reverted, please let me know if that changes anything about how I should structure this PR and I would be happy to make the changes.